src/AdvApi32.Desktop/AdvApi32.cs

```diff
@@ -218,7 +263,7 @@ public static partial class AdvApi32
         /// </returns>
         [DllImport(api_ms_win_service_management_l2_1_0, SetLastError = true, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.Bool)]
-        public static unsafe extern bool ChangeServiceConfig2(SafeServiceHandle hService, ServiceInfoLevel dwInfoLevel, void* lpInfo);
+        public static extern bool ChangeServiceConfig2(SafeServiceHandle hService, ServiceInfoLevel dwInfoLevel, IntPtr lpInfo);
```

Comment:
Take a look on the errors on the generated functions:

The errors are on the declarations of void type. probably because of it having `FriendlyFlags.Optional` and being of `void*` type.

```
        public static unsafe uint GetSecurityInfo(IntPtr handle, SE_OBJECT_TYPE ObjectType, SECURITY_INFORMATION SecurityInfo, ref void? ppsidOwner, ref void? ppsidGroup, ref Kernel32.ACL? ppDacl, ref Kernel32.ACL? ppSacl, ref SECURITY_DESCRIPTOR? ppSecurityDescriptor)
        {
            void ppsidOwnerLocal = ppsidOwner.HasValue ? ppsidOwner.Value : default(void);
            void ppsidGroupLocal = ppsidGroup.HasValue ? ppsidGroup.Value : default(void);
            Kernel32.ACL ppDaclLocal = ppDacl.HasValue ? ppDacl.Value : default(Kernel32.ACL);
            Kernel32.ACL ppSaclLocal = ppSacl.HasValue ? ppSacl.Value : default(Kernel32.ACL);
            SECURITY_DESCRIPTOR ppSecurityDescriptorLocal = ppSecurityDescriptor.HasValue ? ppSecurityDescriptor.Value : default(SECURITY_DESCRIPTOR);
            uint result = GetSecurityInfo(handle, ObjectType, SecurityInfo, ppsidOwner.HasValue ? &ppsidOwnerLocal : null, ppsidGroup.HasValue ? &ppsidGroupLocal : null, ppDacl.HasValue ? &ppDaclLocal : null, ppSacl.HasValue ? &ppSaclLocal : null, ppSecurityDescriptor.HasValue ? &ppSecurityDescriptorLocal : null);
            if (ppsidOwner.HasValue)
                ppsidOwner = ppsidOwnerLocal;
            if (ppsidGroup.HasValue)
                ppsidGroup = ppsidGroupLocal;
            if (ppDacl.HasValue)
                ppDacl = ppDaclLocal;
            if (ppSacl.HasValue)
                ppSacl = ppSaclLocal;
            if (ppSecurityDescriptor.HasValue)
                ppSecurityDescriptor = ppSecurityDescriptorLocal;
            return result;
        }

public static unsafe Win32ErrorCode GetNamedSecurityInfo(string pObjectName, SE_OBJECT_TYPE ObjectType, SECURITY_INFORMATION SecurityInfo, ref void? ppsidOwner, ref void? ppsidGroup, ref Kernel32.ACL? ppDacl, ref Kernel32.ACL? ppSacl, ref SECURITY_DESCRIPTOR? ppSecurityDescriptor)
        {
            void ppsidOwnerLocal = ppsidOwner.HasValue ? ppsidOwner.Value : default(void);
            void ppsidGroupLocal = ppsidGroup.HasValue ? ppsidGroup.Value : default(void);
            Kernel32.ACL ppDaclLocal = ppDacl.HasValue ? ppDacl.Value : default(Kernel32.ACL);
            Kernel32.ACL ppSaclLocal = ppSacl.HasValue ? ppSacl.Value : default(Kernel32.ACL);
            SECURITY_DESCRIPTOR ppSecurityDescriptorLocal = ppSecurityDescriptor.HasValue ? ppSecurityDescriptor.Value : default(SECURITY_DESCRIPTOR);
            Win32ErrorCode result = GetNamedSecurityInfo(pObjectName, ObjectType, SecurityInfo, ppsidOwner.HasValue ? &ppsidOwnerLocal : null, ppsidGroup.HasValue ? &ppsidGroupLocal : null, ppDacl.HasValue ? &ppDaclLocal : null, ppSacl.HasValue ? &ppSaclLocal : null, ppSecurityDescriptor.HasValue ? &ppSecurityDescriptorLocal : null);
            if (ppsidOwner.HasValue)
                ppsidOwner = ppsidOwnerLocal;
            if (ppsidGroup.HasValue)
                ppsidGroup = ppsidGroupLocal;
            if (ppDacl.HasValue)
                ppDacl = ppDaclLocal;
            if (ppSacl.HasValue)
                ppSacl = ppSaclLocal;
            if (ppSecurityDescriptor.HasValue)
                ppSecurityDescriptor = ppSecurityDescriptorLocal;
            return result;
        }

public static unsafe uint SetSecurityInfo(IntPtr handle, SE_OBJECT_TYPE ObjectType, SECURITY_INFORMATION SecurityInfo, ref void? psidOwner, ref void? psidGroup, ref Kernel32.ACL? pDacl, ref Kernel32.ACL? pSacl)
        {
            void psidOwnerLocal = psidOwner.HasValue ? psidOwner.Value : default(void);
            void psidGroupLocal = psidGroup.HasValue ? psidGroup.Value : default(void);
            Kernel32.ACL pDaclLocal = pDacl.HasValue ? pDacl.Value : default(Kernel32.ACL);
            Kernel32.ACL pSaclLocal = pSacl.HasValue ? pSacl.Value : default(Kernel32.ACL);
            uint result = SetSecurityInfo(handle, ObjectType, SecurityInfo, psidOwner.HasValue ? &psidOwnerLocal : null, psidGroup.HasValue ? &psidGroupLocal : null, pDacl.HasValue ? &pDaclLocal : null, pSacl.HasValue ? &pSaclLocal : null);
            if (psidOwner.HasValue)
                psidOwner = psidOwnerLocal;
            if (psidGroup.HasValue)
                psidGroup = psidGroupLocal;
            if (pDacl.HasValue)
                pDacl = pDaclLocal;
            if (pSacl.HasValue)
                pSacl = pSaclLocal;
            return result;
        }

public static unsafe uint SetNamedSecurityInfo(string pObjectName, SE_OBJECT_TYPE ObjectType, SECURITY_INFORMATION SecurityInfo, ref void? psidOwner, ref void? psidGroup, ref Kernel32.ACL? pDacl, ref Kernel32.ACL? pSacl)
        {
            void psidOwnerLocal = psidOwner.HasValue ? psidOwner.Value : default(void);
            void psidGroupLocal = psidGroup.HasValue ? psidGroup.Value : default(void);
            Kernel32.ACL pDaclLocal = pDacl.HasValue ? pDacl.Value : default(Kernel32.ACL);
            Kernel32.ACL pSaclLocal = pSacl.HasValue ? pSacl.Value : default(Kernel32.ACL);
            uint result = SetNamedSecurityInfo(pObjectName, ObjectType, SecurityInfo, psidOwner.HasValue ? &psidOwnerLocal : null, psidGroup.HasValue ? &psidGroupLocal : null, pDacl.HasValue ? &pDaclLocal : null, pSacl.HasValue ? &pSaclLocal : null);
            if (psidOwner.HasValue)
                psidOwner = psidOwnerLocal;
            if (psidGroup.HasValue)
                psidGroup = psidGroupLocal;
            if (pDacl.HasValue)
                pDacl = pDaclLocal;
            if (pSacl.HasValue)
                pSacl = pSaclLocal;
            return result;
        }
```